prog/minimization.go

```diff
@@ -298,7 +298,7 @@ func (typ *ResourceType) minimize(ctx *minimizeArgsCtx, arg Arg, path string) bo
 }
 
 func (typ *BufferType) minimize(ctx *minimizeArgsCtx, arg Arg, path string) bool {
-	if arg.Dir() == DirOut {
+	if arg.Dir() == DirOut || typ.IsCompressed() {
```

Comment:
You already required the call to be non-minimizable. Maybe panic if we are still forced to minimize it here?